contacts/urls.py

```diff
@@ -1,13 +1,14 @@
 from django.conf.urls import url
 from django.contrib.auth.decorators import login_required
 
+from contacts.decorators import user_added_contact
 from . import views
 
 
 app_name = 'contacts'
 urlpatterns = [
     url(r'^contact/?$', login_required(views.ContactsListView.as_view()), name='contact_list'),
-    url(r'^(?P<pk>[0-9]+)/?$', login_required(views.ContactDetailView.as_view()), name='contact_detail'),
+    url(r'^(?P<pk>[0-9]+)/?$', login_required(user_added_contact(views.ContactDetailView.as_view())), name='contact_detail'),
```

Comment:
go through all the other urls and add this new decorator to all the views specific to a user.